clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -201,6 +204,8 @@ protected boolean needRejoin() {
      * Reset the generation/memberId tracked by this member
      */
     public void resetGeneration() {
+        // leave the group if it is needed, but don't block
+        maybeLeaveGroup(0);
```

Comment:
Am I misreading this or will this, combined with the `TimeoutException` from `maybeLeaveGroup` cause this to always throw a `TimeoutException`? Some tests seem to be throwing it, so I think this might be the case and not the behavior you want here.